{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "86f5f900",
   "metadata": {},
   "source": [
    "# Data download\n",
    "\n",
    "Description: Download all external data required to execute the code for Chapters 2-4. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d9f06365",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !pip install urllib3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "5a9dcab1",
   "metadata": {},
   "outputs": [],
   "source": [
    "import os \n",
    "import shutil, gzip\n",
    "import zipfile, tarfile\n",
    "from urllib import request"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "7c6d8e7c",
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_local_data_path(folders, fname=None):\n",
    "    folder_structure = 'external_data/'+'/'.join(folders)\n",
    "    os.makedirs(os.path.normpath(folder_structure), exist_ok=True) \n",
    "    if(fname!=None):\n",
    "        return(os.path.normpath(folder_structure +'/'+ fname))\n",
    "    else:\n",
    "        return(os.path.normpath(folder_structure +'/'))\n",
    "\n",
    "def extract_file(fname, destination_fname):\n",
    "    if fname.endswith(\"tar.gz\"):\n",
    "        tar = tarfile.open(fname, \"r:gz\")\n",
    "        tar.extractall(destination_fname)\n",
    "        tar.close()\n",
    "    elif fname.endswith(\"tar\"):\n",
    "        tar = tarfile.open(fname, \"r:\")\n",
    "        tar.extractall(destination_fname)\n",
    "        tar.close()\n",
    "    elif fname.endswith(\".gz\"):\n",
    "        with gzip.open(fname, 'rb') as f_in:\n",
    "            with open(destination_fname, 'wb') as f_out:\n",
    "                shutil.copyfileobj(f_in, f_out)\n",
    "    elif fname.endswith(\"zip\"):\n",
    "        with zipfile.ZipFile(fname) as zf:\n",
    "            zf.extractall(destination_fname)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "08d79075",
   "metadata": {},
   "source": [
    "## Chapter 2: \n",
    "\n",
    "#### Experimental reproducibility limits the correlation between mRNA and protein abundances in tumour profiles"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "7ca21eeb",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('external_data\\\\mi_yang_25Sept2020\\\\guan_sc2_ovarian_cor.txt',\n",
       " <http.client.HTTPMessage at 0x25d99262940>)"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "## 1. Colorectal Cancer (2014)\n",
    "##    A. Transcriptomics\n",
    "request.urlretrieve(\"https://cbioportal-datahub.s3.amazonaws.com/coadread_tcga_pub.tar.gz\", \n",
    "              get_local_data_path([\"tumour_studies\", \"crc\"], \"coadread_tcga_pub.tar.gz\"))\n",
    "extract_file(get_local_data_path([\"tumour_studies\", \"crc\"], \"coadread_tcga_pub.tar.gz\"), \n",
    "             get_local_data_path([\"tumour_studies\", \"crc\"]))\n",
    "\n",
    "##    B. Proteomics \n",
    "request.urlretrieve(\"https://static-content.springer.com/esm/art%3A10.1038%2Fnature13438/MediaObjects/41586_2014_BFnature13438_MOESM5_ESM.xlsx\", \n",
    "              get_local_data_path([\"tumour_studies\", \"crc\", \"zhang_2014\"],\n",
    "                                        \"Supplemental_Table.xlsx\"))\n",
    "\n",
    "\n",
    "## 2. Breast Cancer (2016)\n",
    "##    A. Transcriptomics\n",
    "request.urlretrieve(\"https://cbioportal-datahub.s3.amazonaws.com/brca_tcga_pub2015.tar.gz\", \n",
    "              get_local_data_path([\"tumour_studies\", \"brca\"], \"brca_tcga_pub2015.tar.gz\"))\n",
    "extract_file(get_local_data_path([\"tumour_studies\", \"brca\"], \"brca_tcga_pub2015.tar.gz\"), \n",
    "             get_local_data_path([\"tumour_studies\", \"brca\"]))\n",
    "\n",
    "##    B. Proteomics \n",
    "request.urlretrieve(\"https://static-content.springer.com/esm/art%3A10.1038%2Fnature18003/MediaObjects/41586_2016_BFnature18003_MOESM111_ESM.zip\", \n",
    "              get_local_data_path([\"tumour_studies\", \"brca\", \"mertins_2016\"], \"BrCa_2016_Supplemental_Tables.zip\"))\n",
    "\n",
    "extract_file(get_local_data_path([\"tumour_studies\", \"brca\", \"mertins_2016\"], \"BrCa_2016_Supplemental_Tables.zip\"), \n",
    "             get_local_data_path([\"tumour_studies\", \"brca\", \"mertins_2016\"]))\n",
    "\n",
    "shutil.move(get_local_data_path([\"tumour_studies\", \"brca\", \"mertins_2016\", \"nature18003-s2\"], \n",
    "                                \"CPTAC_BC_SupplementaryTable03.xlsx\"), \n",
    "            get_local_data_path([\"tumour_studies\", \"brca\", \"mertins_2016\"], \n",
    "                                \"CPTAC_BC_SupplementaryTable03.xlsx\"))\n",
    "\n",
    "## 3. Ovarian Cancer (2016)\n",
    "##    A. Transcriptomics\n",
    "request.urlretrieve(\"http://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/OV/20160128/gdac.broadinstitute.org_OV.mRNA_Preprocess_Median.Level_3.2016012800.0.0.tar.gz\", \n",
    "              get_local_data_path([\"tumour_studies\", \"ovca\"], \"ovca_tcga_pub2015.tar.gz\"))\n",
    "extract_file(get_local_data_path([\"tumour_studies\", \"ovca\"], \"ovca_tcga_pub2015.tar.gz\"), \n",
    "             get_local_data_path([\"tumour_studies\", \"ovca\"]))\n",
    "\n",
    "##    B. Proteomics \n",
    "request.urlretrieve(\"http://gdac.broadinstitute.org/runs/stddata__2016_01_28/data/OV/20160128/gdac.broadinstitute.org_OV.mRNA_Preprocess_Median.Level_3.2016012800.0.0.tar.gz\", \n",
    "              get_local_data_path([\"tumour_studies\", \"ovca\", \"firebrowse\"], \"OV.mRNA_Preprocess_Median.Level_3.2016012800.0.0.tar.gz\"))\n",
    "\n",
    "extract_file(get_local_data_path([\"tumour_studies\", \"ovca\", \"firebrowse\"], \"OV.mRNA_Preprocess_Median.Level_3.2016012800.0.0.tar.gz\"), \n",
    "             get_local_data_path([\"tumour_studies\", \"ovca\", \"firebrowse\"], \"OV.mRNA\"))\n",
    "shutil.move(get_local_data_path([\"tumour_studies\", \"ovca\", \"firebrowse\", \"OV.mRNA\", \n",
    "                                \"gdac.broadinstitute.org_OV.mRNA_Preprocess_Median.Level_3.2016012800.0.0\"], \n",
    "                                \"OV.medianexp.txt\"), \n",
    "            get_local_data_path([\"tumour_studies\", \"ovca\", \"firebrowse\"], \"OV.medianexp.txt\"))\n",
    "\n",
    "## 4. Colon Adenocarcinoma (2019)\n",
    "##    A. Transcriptomics\n",
    "request.urlretrieve(\"http://linkedomics.org/cptac-colon/Human__CPTAC_COAD__UNC__RNAseq__HiSeq_RNA__03_01_2017__BCM__Gene__BCM_RSEM_UpperQuartile_log2.cct.gz\", \n",
    "              get_local_data_path([\"tumour_studies\", \"colon\", \"vasaikar_2019\"], \"RNAseq.cct\"))\n",
    "\n",
    "##    B. Proteomics \n",
    "##       TMT 10-plex Mass spectrometry data:\n",
    "request.urlretrieve(\"http://linkedomics.org/cptac-colon/Human__CPTAC_COAD__PNNL__Proteome__TMT__03_01_2017__BCM__Gene__Tumor_Normal_log2FC.cct\", \n",
    "              get_local_data_path([\"tumour_studies\", \"colon\", \"vasaikar_2019\"], \"Proteome_TMT.cct\"))\n",
    "\n",
    "##       LF Mass spectrometry data\n",
    "request.urlretrieve(\"http://linkedomics.org/cptac-colon/Human__CPTAC_COAD__VU__Proteome__QExact__03_01_2017__BCM__Gene__VU_Tumor_LF_UnsharedCounts.cct\", \n",
    "              get_local_data_path([\"tumour_studies\", \"colon\", \"vasaikar_2019\"], \"Proteome_LF.cct\"))\n",
    "\n",
    "\n",
    "## 5. Cancer Cell Lines Encyclopedia (2020)\n",
    "##    A. Transcriptomics\n",
    "request.urlretrieve(\"https://ndownloader.figshare.com/files/25494389\", \n",
    "              get_local_data_path([\"depmap\", \"20Q4\"], \"CCLE_expression.csv\"))\n",
    "\n",
    "##    B. Cell Line Mapping File \n",
    "request.urlretrieve(\"https://ndownloader.figshare.com/files/25494443\", \n",
    "              get_local_data_path([\"depmap\", \"20Q4\"], \"sample.csv\"))\n",
    "\n",
    "##    C. For processing transcriptomic replicates:\n",
    "request.urlretrieve(\"https://static-content.springer.com/esm/art%3A10.1038%2Fs41586-019-1186-3/MediaObjects/41586_2019_1186_MOESM4_ESM.xlsx\", \n",
    "              get_local_data_path([\"ghandi_2019\"], \"Table_S1.xlsx\"))\n",
    "\n",
    "##    D. Proteomics Data\n",
    "request.urlretrieve(\"https://gygi.hms.harvard.edu/data/ccle/Table_S2_Protein_Quant_Normalized.xlsx\", \n",
    "              get_local_data_path([\"nusinow_2020\"], \"Table_S2.xlsx\"))\n",
    "\n",
    "##    E. Protein expression levels in biological replicates\n",
    "request.urlretrieve(\"https://gygi.hms.harvard.edu/data/ccle/Table_S3_Biological_Replicates_Protein_Quant_Normalized.xlsx\", \n",
    "              get_local_data_path([\"nusinow_2020\"], \"Table_S3.xlsx\"))\n",
    "\n",
    "## The commented code indicate the urls in those lines cannot be accessed programmatically and \n",
    "## will have to be downloaded manually! \n",
    "\n",
    "## 6. NCI60 cancer cell lines (2019)\n",
    "##    A. Transcriptomics data \n",
    "\n",
    "# request.urlretrieve(\"https://www.cell.com/cms/10.1016/j.isci.2019.10.059/attachment/816cfb48-4f4d-4ace-843b-e4300bd85b7a/mmc7.xlsx\", \n",
    "#               get_local_data_path([\"guo_2019\"], \"Table_S6.xlsx\"))\n",
    "\n",
    "##    B. Proteomics data \n",
    "# request.urlretrieve(\"https://www.cell.com/cms/10.1016/j.isci.2019.10.059/attachment/30dfcf80-68fd-4f82-aef4-e27ab8bacc47/mmc2.xlsx\", \n",
    "#               get_local_data_path([\"guo_2019\"], \"Table S1.xlsx\"))\n",
    "\n",
    "# ## 7. GTEx 32 Healthy Tissues (2020)\n",
    "# ##    A. Transcriptomics Data\n",
    "# request.urlretrieve(\"https://www.cell.com/cms/10.1016/j.cell.2020.08.036/attachment/bab7713d-6ee6-4d19-badc-aceb88535518/mmc4.xlsx\", \n",
    "#               get_local_data_path([\"jiang_2020\"], \"Table_S4.xlsx\"))\n",
    "\n",
    "# ##    B. Proteomics Data\n",
    "# request.urlretrieve(\"https://www.cell.com/cms/10.1016/j.cell.2020.08.036/attachment/0d79a576-f9b7-4342-81a8-f80c14df0372/mmc3.xlsx\", \n",
    "#               get_local_data_path([\"jiang_2020\"], \"Table_S3.xlsx\"))\n",
    "\n",
    "# ##    C. Experiment Info\n",
    "# request.urlretrieve(\"https://www.cell.com/cms/10.1016/j.cell.2020.08.036/attachment/e874e52c-b2b9-4771-8ceb-9415fed1750b/mmc5.xlsx\", \n",
    "#               get_local_data_path([\"jiang_2020\"], \"Table_S5.xlsx\"))\n",
    "\n",
    "## 8. Protein half-lives\n",
    "request.urlretrieve(\"https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5930408/bin/supp_RA118.000583_134915_1_supp_65961_p3f7xq.xlsx\", \n",
    "              get_local_data_path([\"protein_properties\", \"zecha_2018\"], \"Table_S3.xlsx\"))\n",
    "\n",
    "## 9. Protein Complexes\n",
    "request.urlretrieve(\"https://mips.helmholtz-muenchen.de/corum/download/releases/old/corum_2018_09_03.zip\", \n",
    "              get_local_data_path([\"protein_properties\", \"CORUM\"], \"allComplexes.zip\"))\n",
    "extract_file(get_local_data_path([\"protein_properties\", \"CORUM\"], \"allComplexes.zip\"), \n",
    "             get_local_data_path([\"protein_properties\", \"CORUM\"]))\n",
    "extract_file(get_local_data_path([\"protein_properties\", \"CORUM\", \"current\"], \"allComplexes.txt.zip\"), \n",
    "             get_local_data_path([\"protein_properties\", \"CORUM\"]))\n",
    "\n",
    "## 10. NCI CPTAC DREAM Challenge best performer's prediction score \n",
    "# Received from authors directly.\n",
    "request.urlretrieve(\"https://raw.githubusercontent.com/SwathiRUpadhya/PhD_thesis/main/other_data/mi_yang_25Sept2020/guan_sc2_breast_cor.txt\", \n",
    "              get_local_data_path([\"mi_yang_25Sept2020\"], \"guan_sc2_breast_cor.txt\"))\n",
    "request.urlretrieve(\"https://raw.githubusercontent.com/SwathiRUpadhya/PhD_thesis/main/other_data/mi_yang_25Sept2020/guan_sc2_ovarian_cor.txt\", \n",
    "              get_local_data_path([\"mi_yang_25Sept2020\"], \"guan_sc2_ovarian_cor.txt\"))\n",
    "\n",
    "## 11. Transcriptomics data of 675 human cancer cell lines\n",
    "# request.urlretrieve(\"https://www.ebi.ac.uk/gxa/experiments-content/E-MTAB-2706/resources/ExperimentDownloadSupplier.RnaSeqBaseline/tpms.tsv\", \n",
    "#               get_local_data_path([\"klijn_2015\"], \"E-MTAB-2706-query-results.tpms.tsv\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "3f2c2e15",
   "metadata": {},
   "source": [
    "## Chapter 3: \n",
    "#### Anitbody reliability limits the correlation between mRNA and protein abundances in tumours "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5d5d38dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('external_data\\\\depmap\\\\ProteinArray_RPPA\\\\CCLE_RPPA_Ab_info_20180123.csv',\n",
       " <http.client.HTTPMessage at 0x25d99251580>)"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# 1. Antibody validation status\n",
    "#    A. TCPA antibody validation status\n",
    "request.urlretrieve(\"https://static-content.springer.com/esm/art%3A10.1038%2Fnmeth.2650/MediaObjects/41592_2013_BFnmeth2650_MOESM330_ESM.xlsx\", \n",
    "              get_local_data_path(['jun_li_2013'], 'nmeth.2650-S2.xlsx'))\n",
    "\n",
    "#    B. MD Anderson Cancer Center:\n",
    "request.urlretrieve(\"https://www.mdanderson.org/content/dam/mdanderson/documents/core-facilities/Functional%20Proteomics%20RPPA%20Core%20Facility/RPPA_Standard_Ab_List_Updated.xlsx\", \n",
    "              get_local_data_path(['mdanderson_cancer_center'], 'RPPA_Standard_Ab_List_Updated.xlsx'))\n",
    "\n",
    "#    C. MD Anderson Cancer Center - Corrected Gene Names\n",
    "request.urlretrieve(\"https://www.mdanderson.org/content/dam/mdanderson/documents/core-facilities/Functional%20Proteomics%20RPPA%20Core%20Facility/Corrected_Gene_Names.pdf\", \n",
    "              get_local_data_path(['mdanderson_cancer_center'], 'Corrected_Gene_Names.pdf'))\n",
    "\n",
    "# 2. Tumour studies\n",
    "#    A. TCGA transcriptomics:\n",
    "request.urlretrieve(\"https://tcga-pancan-atlas-hub.s3.us-east-1.amazonaws.com/download/EB%2B%2BAdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena.gz\", \n",
    "              get_local_data_path(['usc_xena_browser'], 'EB++AdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena.gz'))\n",
    "extract_file(get_local_data_path(['usc_xena_browser'], 'EB++AdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena.gz'), \n",
    "             get_local_data_path(['usc_xena_browser'], 'EB++AdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena'))\n",
    "\n",
    "#    B. TCGA proteomics:\n",
    "request.urlretrieve(\"https://tcga-pancan-atlas-hub.s3.us-east-1.amazonaws.com/download/TCGA-RPPA-pancan-clean.xena.gz\", \n",
    "              get_local_data_path(['usc_xena_browser'], 'TCGA-RPPA-pancan-clean.xena.gz'))\n",
    "extract_file(get_local_data_path(['usc_xena_browser'], 'TCGA-RPPA-pancan-clean.xena.gz'), \n",
    "             get_local_data_path(['usc_xena_browser'], 'TCGA-RPPA-pancan-clean.xena')), \n",
    "             \n",
    "\n",
    "#    C. Mapping samples through GDC Reference file\n",
    "request.urlretrieve(\"http://api.gdc.cancer.gov/data/fcbb373e-28d4-4818-92f3-601ede3da5e1\", \n",
    "              get_local_data_path(['gdc_cancer_gov'], 'TCGA-RPPA-pancan-clean.txt'))\n",
    "\n",
    "# 3. Cancer Cell Lines Encyclopedia (2020)\n",
    "#    A. Transcriptomics\n",
    "request.urlretrieve(\"https://ndownloader.figshare.com/files/25494389\", \n",
    "              get_local_data_path([\"depmap\", \"20Q4\"], \"CCLE_expression.csv\"))\n",
    "\n",
    "#    B. Cell Line Mapping File \n",
    "request.urlretrieve(\"https://ndownloader.figshare.com/files/25494443\", \n",
    "              get_local_data_path([\"depmap\", \"20Q4\"], \"sample.csv\"))\n",
    "\n",
    "#    C. Proteomics Data - Mass spec\n",
    "request.urlretrieve(\"https://gygi.hms.harvard.edu/data/ccle/Table_S2_Protein_Quant_Normalized.xlsx\", \n",
    "              get_local_data_path([\"nusinow_2020\"], \"Table_S2.xlsx\"))\n",
    "\n",
    "#    D. Proteomics Data - RPPA data\n",
    "request.urlretrieve(\"https://depmap.org/portal/download/api/download?file_name=ccle%2FCCLE_RPPA_20180123.csv&bucket=depmap-external-downloads\", \n",
    "              get_local_data_path([\"depmap\", \"ProteinArray_RPPA\"], \"CCLE_RPPA_20180123.csv\"))\n",
    "\n",
    "#    E. Antibody validation status file\n",
    "request.urlretrieve(\"https://depmap.org/portal/download/api/download?file_name=ccle%2FCCLE_RPPA_Ab_info_20180123.csv&bucket=depmap-external-downloads\", \n",
    "              get_local_data_path([\"depmap\", \"ProteinArray_RPPA\"], \"CCLE_RPPA_Ab_info_20180123.csv\"))\n",
    "\n",
    "### same files are downloaded in Project 1 already. \n",
    "# ## 4. GTEx 32 Healthy Tissues (2020)\n",
    "# ##    A. Transcriptomics Data\n",
    "# request.urlretrieve(\"https://www.cell.com/cms/10.1016/j.cell.2020.08.036/attachment/bab7713d-6ee6-4d19-badc-aceb88535518/mmc4.xlsx\", \n",
    "#               get_local_data_path([\"jiang_2020\"], \"Table_S4.xlsx\"))\n",
    "\n",
    "# ##    B. Proteomics Data\n",
    "# request.urlretrieve(\"https://www.cell.com/cms/10.1016/j.cell.2020.08.036/attachment/0d79a576-f9b7-4342-81a8-f80c14df0372/mmc3.xlsx\", \n",
    "#               get_local_data_path([\"jiang_2020\"], \"Table_S3.xlsx\"))\n",
    "\n",
    "## difficulty in accessing the data programmatically\n",
    "# 5. Aggregated Protein Reproducibility\n",
    "# request.urlretrieve(\"https://www.cell.com/cell-reports-methods/fulltext/S2667-2375(22)00170-9\", \n",
    "#               get_local_data_path([\"upadhya_ryan_2022\"], \"TableS2.xlsx\"))"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "103804f3",
   "metadata": {},
   "source": [
    "## Chapter 4: \n",
    "#### Latent space models aid in the prediction of protein abundances in tumours"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "99d63ed8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "('external_data\\\\hgnc_22Feb2022\\\\approved_protein_coding_genes.txt',\n",
       " <http.client.HTTPMessage at 0x25d992d2130>)"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#### 1. Linkedomics Transcriptomics Data: \n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/TCGA-BRCA/Human__TCGA_BRCA__UNC__RNAseq__HiSeq_RNA__01_28_2016__BI__Gene__Firehose_RSEM_log2.cct.gz\", \n",
    "              get_local_data_path(['transcriptomics_data'], 'TCGA_BRCA_RNAseq_RSEM_log2.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/TCGA-OV/Human__TCGA_OV__UNC__RNAseq__HiSeq_RNA__01_28_2016__BI__Gene__Firehose_RSEM_log2.cct.gz\", \n",
    "              get_local_data_path(['transcriptomics_data'], 'TCGA_OV_RNAseq_RSEM_log2.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-PDAC/mRNA_RSEM_UQ_log2_Tumor.cct\", \n",
    "              get_local_data_path(['transcriptomics_data'], 'CPTAC_PDAC_mRNA_RSEM_UQ_log2_Tumor.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-BRCA/HS_CPTAC_BRCA_2018_RNA_GENE.cct\", \n",
    "              get_local_data_path(['transcriptomics_data'], 'HS_CPTAC_BRCA_2018_RNA_GENE.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-OV/HS_CPTAC_OV_rnaseq_fpkm_log2.cct\", \n",
    "              get_local_data_path(['transcriptomics_data'], 'HS_CPTAC_OV_rnaseq_fpkm_log2.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-CCRCC/HS_CPTAC_CCRCC_RNAseq_fpkm_log2_Tumor.cct\", \n",
    "              get_local_data_path(['transcriptomics_data'], 'HS_CPTAC_CCRCC_RNAseq_fpkm_log2_Tumor.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-LUAD/HS_CPTAC_LUAD_rnaseq_uq_rpkm_log2_NArm_TUMOR.cct\", \n",
    "              get_local_data_path(['transcriptomics_data'], 'HS_CPTAC_LUAD_rnaseq_uq_rpkm_log2_NArm_TUMOR.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-LSCC/HS_CPTAC_LSCC_2020_rnaseq_uq_fpkm_log2_NArm_TUMOR.cct\", \n",
    "              get_local_data_path(['transcriptomics_data'], 'HS_CPTAC_LSCC_2020_rnaseq_uq_fpkm_log2_NArm_TUMOR.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-UCEC/HS_CPTAC_UCEC_RNAseq_RSEM_UQ_log2_Tumor.cct\", \n",
    "              get_local_data_path(['transcriptomics_data'], 'HS_CPTAC_UCEC_RNAseq_RSEM_UQ_log2_Tumor.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-GBM/HS_CPTAC_GBM_rnaseq_fpkm_uq_log2.cct\", \n",
    "              get_local_data_path(['transcriptomics_data'], 'HS_CPTAC_GBM_rnaseq_fpkm_uq_log2.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-HNSCC/HS_CPTAC_HNSCC_RNAseq_RSEM_UQ_log2_Tumor.cct\", \n",
    "              get_local_data_path(['transcriptomics_data'], 'HS_CPTAC_HNSCC_RNAseq_RSEM_UQ_log2_Tumor.cct'))\n",
    "\n",
    "\n",
    "\n",
    "## 2. Linkedomics Proteomics \n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-BRCA/HS_CPTAC_BRCA_2018_Proteome_Ratio_Norm_gene_Median.cct\", \n",
    "              get_local_data_path(['proteomics_data'], 'CPTAC_BRCA_2018_Proteome_gene_Median.cct'))\n",
    "\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-OV/HS_CPTAC_OV_proteome_gene_tumor.cct\", \n",
    "              get_local_data_path(['proteomics_data'], 'CPTAC_OV_proteome_gene_tumor.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-CCRCC/HS_CPTAC_CCRCC_proteome_Tumor.cct\", \n",
    "              get_local_data_path(['proteomics_data'], 'CPTAC_CCRCC_proteome_Tumor.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-LUAD/HS_CPTAC_LUAD_proteome_ratio_NArm_TUMOR.cct\", \n",
    "              get_local_data_path(['proteomics_data'], 'CPTAC_LUAD_proteome_ratio_NArm_TUMOR.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-LSCC/HS_CPTAC_LSCC_2020_proteome_ratio_NArm_TUMOR.cct\", \n",
    "              get_local_data_path(['proteomics_data'], 'CPTAC_LSCC_2020_proteome_ratio_NArm_TUMOR.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-UCEC/HS_CPTAC_UCEC_Proteomics_TMT_gene_level_Tumor.cct\", \n",
    "              get_local_data_path(['proteomics_data'], 'CPTAC_UCEC_Proteomics_TMT_gene_level_Tumor.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-GBM/HS_CPTAC_GBM_proteome_mssm_per_gene.cct\", \n",
    "              get_local_data_path(['proteomics_data'], 'CPTAC_GBM_proteome_mssm_per_gene.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-HNSCC/HS_CPTAC_HNSCC_Proteomics_TMT_Gene_level_Tumor.cct\", \n",
    "              get_local_data_path(['proteomics_data'], 'CPTAC_HNSCC_Proteomics_TMT_Gene_level_Tumor.cct'))\n",
    "\n",
    "request.urlretrieve(\"https://linkedomics.org/data_download/CPTAC-PDAC/proteomics_gene_level_MD_abundance_tumor.cct\", \n",
    "              get_local_data_path(['tumour_studies','pdac'], 'proteomics_gene_level_MD_abundance_tumor.cct.txt'))\n",
    "\n",
    "\n",
    "## already downloaded\n",
    "# request.urlretrieve(\"https://static-content.springer.com/esm/art%3A10.1038%2Fnature18003/MediaObjects/41586_2016_BFnature18003_MOESM111_ESM.zip\", \n",
    "#               get_local_data_path([\"tumour_studies\", \"brca\", \"mertins_2016\"], \"BrCa_2016_Supplemental_Tables.zip\"))\n",
    "\n",
    "# extract_file(get_local_data_path([\"tumour_studies\", \"brca\", \"mertins_2016\"], \"BrCa_2016_Supplemental_Tables.zip\"), \n",
    "#              get_local_data_path([\"tumour_studies\", \"brca\", \"mertins_2016\"]))\n",
    "\n",
    "# shutil.move(get_local_data_path([\"tumour_studies\", \"brca\", \"mertins_2016\", \"nature18003-s2\"], \n",
    "#                                 \"CPTAC_BC_SupplementaryTable03.xlsx\"), \n",
    "#             get_local_data_path([\"tumour_studies\", \"brca\", \"mertins_2016\"], \n",
    "#                                 \"CPTAC_BC_SupplementaryTable03.xlsx\"))\n",
    "\n",
    "## Returns error in programmatic access of data, so needs to be downloaded manually\n",
    "# request.urlretrieve(\"https://www.cell.com/cms/10.1016/j.cell.2016.05.069/attachment/15e46617-bec0-42cc-82fb-71f842e8aaac/mmc3.xlsx\", \n",
    "#               get_local_data_path(['tumour_studies','ovca', 'zhang_2016'], 'Table_S2.xlsx'))\n",
    "\n",
    "## 3. Sample Info\n",
    "request.urlretrieve(\"https://raw.githubusercontent.com/SwathiRUpadhya/PhD_thesis/main/other_data/brca_clinical_data/combined_study_clinical_data.tsv\", \n",
    "              get_local_data_path(['tumour_studies','brca', 'mertins_2016'], \n",
    "                                         'combined_study_clinical_data.tsv'))\n",
    "\n",
    "# request.urlretrieve(\"https://www.cell.com/cms/10.1016/j.cell.2016.05.069/attachment/04331724-1bc2-486b-819c-c38972d50492/mmc2.xlsx\", \n",
    "#               get_local_data_path(['tumour_studies','ovca', 'zhang_2016'], 'Table_S1.xlsx'))\n",
    "\n",
    "## 4. UCSC Xena - batch corrected gene expression\n",
    "request.urlretrieve(\"https://xenabrowser.net/datapages/?dataset=EB%2B%2BAdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena&host=https%3A%2F%2Fpancanatlas.xenahubs.net&removeHub=https%3A%2F%2Fucscpublic.xenahubs.net&removeHub=https%3A%2F%2Ftcga.xenahubs.net&removeHub=https%3A%2F%2Ficgc.xenahubs.net&removeHub=https%3A%2F%2Fpcawg.xenahubs.net&removeHub=https%3A%2F%2Ftoil.xenahubs.net&removeHub=https%3A%2F%2Fxena.treehouse.gi.ucsc.edu%3A443&removeHub=https%3A%2F%2Fgdc.xenahubs.net&removeHub=https%3A%2F%2Fatacseq.xenahubs.net&removeHub=https%3A%2F%2Fkidsfirst.xenahubs.net\", \n",
    "                     get_local_data_path(['usc_xena_browser'], 'EB++AdjustPANCAN_IlluminaHiSeq_RNASeqV2.geneExp.xena'))\n",
    "\n",
    "## 5. TCGA survival analysis data\n",
    "request.urlretrieve(\"https://figshare.com/ndownloader/files/41810850\", \n",
    "                    get_local_data_path(['usc_xena_browser'], 'Survival_SupplementalTable_S1_20171025_xena_sp'))\n",
    "\n",
    "## 6. STRING DB:\n",
    "request.urlretrieve(\"https://stringdb-downloads.org/download/protein.links.v11.5/9606.protein.links.v11.5.txt.gz\", \n",
    "                    get_local_data_path(['stringdb'], '9606.protein.links.v11.5.txt.gz'))\n",
    "extract_file(get_local_data_path(['stringdb'], '9606.protein.links.v11.5.txt.gz'), \n",
    "             get_local_data_path(['stringdb'], '9606.protein.links.v11.5.txt'))\n",
    "\n",
    "request.urlretrieve(\"https://stringdb-downloads.org/download/protein.physical.links.v11.5/9606.protein.physical.links.v11.5.txt.gz\", \n",
    "                    get_local_data_path(['stringdb'], '9606.protein.physical.links.v11.5.txt.gz'))\n",
    "extract_file(get_local_data_path(['stringdb'], '9606.protein.physical.links.v11.5.txt.gz'), \n",
    "             get_local_data_path(['stringdb'], '9606.protein.physical.links.v11.5.txt'))\n",
    "\n",
    "request.urlretrieve(\"https://stringdb-downloads.org/download/protein.info.v11.5/9606.protein.info.v11.5.txt.gz\", \n",
    "                    get_local_data_path(['stringdb'], '9606.protein.info.v11.5.txt.gz'))\n",
    "extract_file(get_local_data_path(['stringdb'], '9606.protein.info.v11.5.txt.gz'), \n",
    "             get_local_data_path(['stringdb'], '9606.protein.info.v11.5.txt'))\n",
    "\n",
    "## 7. Approved protein-encoding genes:\n",
    "request.urlretrieve(\"https://raw.githubusercontent.com/SwathiRUpadhya/PhD_thesis/main/other_data/hgnc/approved_protein_coding_genes.txt\", \n",
    "                    get_local_data_path(['hgnc_22Feb2022'], 'approved_protein_coding_genes.txt'))\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.16"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
